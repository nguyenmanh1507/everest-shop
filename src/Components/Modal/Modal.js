// @flow

import React, { Component, Fragment } from 'react'

type Props = {
  children: any,
  onRequestClose: (e: SyntheticEvent<HTMLElement>) => void,
  show?: boolean
}

class Modal extends Component<Props> {
  modal: ?HTMLDivElement

  handleOnWheel = (e: SyntheticWheelEvent<>) => {
    if (this.modal) {
      const delta = e.deltaY
      this.modal.scrollTop += (delta < 0 ? -1 : 1) * 30
      e.preventDefault()
    }
  }

  render() {
    const { onRequestClose, children, show } = this.props

    return (
      <Fragment>
        <div
          className="modal"
          tabIndex="-1"
          role="dialog"
          onWheel={this.handleOnWheel}
          style={{ display: show ? 'block' : null }}
          ref={el => {
            this.modal = el
          }}
        >
          <div className="modal__dialog" role="document">
            <div className="modal__content">
              <div className="modal__body">{children}</div>
              <button
                type="button"
                className="modal__btn-close"
                data-dismiss="modal"
                aria-label="Close"
                onClick={onRequestClose}
              >
                <i className="fa fa-times fa-lg" />
              </button>
            </div>
          </div>
        </div>
        <div
          className="modal-backdrop"
          onClick={onRequestClose}
          onWheel={e => {
            e.preventDefault()
          }}
        />
      </Fragment>
    )
  }
}

export default Modal
