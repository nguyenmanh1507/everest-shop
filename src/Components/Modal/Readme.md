Modal example

```js
// ISSUE: Don't know how to use React's state for trigger Modal
<Modal onRequestClose={() => {}}>
  <h1>Modal content</h1>
</Modal>;
<button className="btn btn--primary">Show modal</button>
```
