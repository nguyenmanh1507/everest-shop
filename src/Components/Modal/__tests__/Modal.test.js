// @flow

import React from 'react'
import renderer from 'react-test-renderer'

import Modal from '../Modal'

const mockFn = jest.fn()

test('Modal render properly', () => {
  const component = renderer.create(
    <Modal onRequestClose={mockFn}>
      <h1>Modal content go here</h1>
    </Modal>
  )

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Modal render default show true', () => {
  const component = renderer.create(
    <Modal onRequestClose={mockFn} show={true}>
      <h1>Modal content go here</h1>
    </Modal>
  )

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
