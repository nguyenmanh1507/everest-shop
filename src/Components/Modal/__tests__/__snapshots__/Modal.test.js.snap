// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal render properly 1`] = `
Array [
  <div
    className="modal"
    onWheel={[Function]}
    role="dialog"
    style={
        Object {
            "display": null,
          }
    }
    tabIndex="-1"
>
    <div
        className="modal__dialog"
        role="document"
    >
        <div
            className="modal__content"
        >
            <div
                className="modal__body"
            >
                <h1>
                    Modal content go here
                </h1>
            </div>
            <button
                aria-label="Close"
                className="modal__btn-close"
                data-dismiss="modal"
                onClick={[Function]}
                type="button"
            >
                <i
                    className="fa fa-times fa-lg"
                />
            </button>
        </div>
    </div>
</div>,
  <div
    className="modal-backdrop"
    onClick={[Function]}
    onWheel={[Function]}
/>,
]
`;
